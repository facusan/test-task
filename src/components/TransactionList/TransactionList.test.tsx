import renderer from "react-test-renderer";
import { Transaction } from "../../services/TransactionsService";
import TransaciontionList from "./TransactionList";

it("renders TransaciontionList correctly", () => {
  let transactions: Transaction[] = [
    {
      date: new Date(),
      from: "#123",
      id: 1,
      to: "#456",
      value: 100,
    },
    {
      date: new Date(),
      from: "#123",
      id: 2,
      to: "#456",
      value: 100,
    },
    {
      date: new Date(),
      from: "#123",
      id: 3,
      to: "#456",
      value: 100,
    },
  ];
  const component = renderer.create(
    <TransaciontionList transactions={transactions}></TransaciontionList>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
