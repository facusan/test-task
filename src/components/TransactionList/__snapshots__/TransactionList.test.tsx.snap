// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TransaciontionList correctly 1`] = `
Array [
  <div
    className="sc-gicCDI jLHDn"
  >
    <div
      className="sc-bczRLJ cbxpUU"
    >
      <div
        className="sc-gsnTZi ctulah"
      >
        <img
          alt="ether"
          src="ether.png"
          style={
            Object {
              "height": "30px",
            }
          }
        />
      </div>
      <div
        className="sc-dkzDqf kEkHqq"
      >
        <div
          className="sc-hKMtZM fJHpc"
        >
          <div>
            Sent Ether
          </div>
          <div>
            -100 ETH
          </div>
        </div>
        <div
          className="sc-eCYdqJ edlkWG"
        >
          <div>
            1-1-1970 1:0
          </div>
          <div>
            -$170000 USD
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="sc-gicCDI jLHDn"
  >
    <div
      className="sc-bczRLJ cbxpUU"
    >
      <div
        className="sc-gsnTZi ctulah"
      >
        <img
          alt="ether"
          src="ether.png"
          style={
            Object {
              "height": "30px",
            }
          }
        />
      </div>
      <div
        className="sc-dkzDqf kEkHqq"
      >
        <div
          className="sc-hKMtZM fJHpc"
        >
          <div>
            Sent Ether
          </div>
          <div>
            -100 ETH
          </div>
        </div>
        <div
          className="sc-eCYdqJ edlkWG"
        >
          <div>
            1-1-1970 1:0
          </div>
          <div>
            -$170000 USD
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="sc-bczRLJ cbxpUU"
    >
      <div
        className="sc-gsnTZi ctulah"
      >
        <img
          alt="ether"
          src="ether.png"
          style={
            Object {
              "height": "30px",
            }
          }
        />
      </div>
      <div
        className="sc-dkzDqf kEkHqq"
      >
        <div
          className="sc-hKMtZM fJHpc"
        >
          <div>
            Sent Ether
          </div>
          <div>
            -100 ETH
          </div>
        </div>
        <div
          className="sc-eCYdqJ edlkWG"
        >
          <div>
            1-1-1970 1:0
          </div>
          <div>
            -$170000 USD
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
